packages/node/src/indexer/api.service.ts

```diff
@@ -26,24 +34,117 @@ const NOT_SUPPORT = (name: string) => () => {
 
 // https://github.com/polkadot-js/api/blob/12750bc83d8d7f01957896a80a7ba948ba3690b7/packages/rpc-provider/src/ws/index.ts#L43
 const RETRY_DELAY = 2_500;
+const MAX_RECONNECT_ATTEMPTS = 5;
 
 const logger = getLogger('api');
 
+export class ApiResponseTimeBuffer extends Queue<number> {
+  constructor(capacity: number) {
+    super(capacity);
+  }
+
+  average() {
+    if (this.items.length === 0) {
+      return 0;
+    }
+    return this.items.reduce((a, b) => a + b, 0) / this.items.length;
+  }
+}
+
+export class ApiLoadBalancer {
```

Comment:
Whats the reason for not using a round robin to determine the next endpoint to use? This over complicates things IMO.

I think we should remove this for now, it doesn't take into consideration whether the endpoint is connected